src/components/SecondaryPanes/Frames.js

```diff
@@ -23,16 +29,25 @@ function renderFrameTitle({ displayName }: Frame) {
   return dom.div({ className: "title" }, endTruncateStr(displayName, 40));
 }
 
-function renderFrameLocation({ source, location }: Frame) {
+function renderFrameLocation({ source, location, library }: LocalFrame) {
   const thisSource: ?Source = source;
   if (thisSource == null) {
     return;
   }
 
+  if (library) {
+    return dom.div({ className: "location" }, library);
+  }
+
   const filename = getFilename(thisSource);
   return dom.div({ className: "location" }, `${filename}: ${location.line}`);
 }
 
+function filterDuplicates(list, predicate) {
+  const pairs = zip(list.slice(1), list.slice(0, -1));
+  return pairs.filter(predicate).map(([prev, item]) => item);
+}
```

Comment:
I think there might be a small mistake here.

Consider the example where `list = [1,2,3,4]` and `predicate = (() => true)`. We might expect that `filterDulicates([1,2,3,4], ()=>true)` should return `[1,2,3,4]` but it instead returns `[1,2,3]`.

So, this might display one less entry in call stack.